client/controllers/contacts.spec.js

```diff
@@ -0,0 +1,45 @@
+describe("The client-side ContactController", function() {
+    // module() is provided by angular-mocks.js
+    beforeEach(module("app"));
+    
+    var $controller, $httpBackend, $rootScope;
+    var contactsController, expectedAllContacts, expectedNewContact;
+    
+    // $controller is the service responsible for instantiating controllers.
+    // It is not available on the global scope, so we must inject it this way.
+    beforeEach(inject(function($injector) {
+        $httpBackend = $injector.get("$httpBackend");
+        $rootScope = $injector.get("$rootScope");
+        $controller = $injector.get("$controller");
+        
+        expectedAllContacts = readJSON("development-data.json").allContacts;
+        expectedNewContact = readJSON("development-data.json").newContact;
+        
+        $httpBackend.expectGET("/contacts").respond(200, expectedAllContacts);
+        contactsController = $controller("ContactController", {"$scope": $rootScope});
+        $httpBackend.flush();
+    }));
+    
+    afterEach(function() {
+        $httpBackend.verifyNoOutstandingExpectation();
+        $httpBackend.verifyNoOutstandingRequest();
+    });
+    
+    it("can get all contacts", function() {
+        $httpBackend.expectGET("/contacts").respond(200, expectedAllContacts);
```

Comment:
This test needs to be cleaned up. Currently, a call to `GET '/contacts'` is being made as soon as the controller is instantiated because I want the contacts list to be updated immediately. This is causing me to test that functionality instead of the actual `getContacts()` function. I may be able to change how the grid is initialized _or_ change this test so that it ignores that first GET request. I'm not sure the latter is best practice.

I could do this on another branch later if need be. It isn't the most important thing in the world.